src/infer_single_paired.py

```diff
@@ -0,0 +1,262 @@
+        match_1 = first_convention.finditer(seq_id)
+        match_2 = second_convention.finditer(seq_id)
+        len_match_1 = None
+        len_match_2 = None
+        len_seq_id = len(seq_id)
+        for match in match_2:
+            len_match_2 = match.end() - match.start()
+            break
+        for match in match_1:
+            len_match_1 = match.end() - match.start()
+            break
+        if len_match_1 is not None:
+            if len_seq_id == len_match_1:
+                return 1
+            else:
+                return 0
+        else:
+            if len_match_2 is None:
+                return 0
+            else:
+                if len_seq_id == len_match_2:
+                    return 2
+                else:
+                    return 0
+
+    def check_file1(self):
+        """Opening and checking errors for first file by the user.
+
+        Attribute Error:
+            Function can raise error if file doesn't exist or
+            something wrong with opening of file.
+
+        """
+        try:
+            _open = partial(
+                gzip.open, mode='rt') if self.file1_name.endswith(".gz") else open
+            self.file1 = _open(self.file1_name)
+        except OSError:
+            print("Could not open/read file:", self.file1_name)
+            sys.exit()
+
+    def check_file2(self):
+        # Opening and checking errors for second file by the user
+        try:
+            _open = partial(
+                gzip.open, mode='rt') if self.file2_name.endswith(".gz") else open
+            self.file2 = _open(self.file2_name)
+        except OSError:
+            print("Could not open/read file:", self.file2_name)
+            sys.exit()
+
+    def read_line_in_fastq(self, seq_id: str, seq_id_dict: dict):
+        """This function processes for every seq_id in a fastq file.
+
+        Args:
+            seq_id (str): Sequence identifier.
+            seq_id_dict (dict) : Dictionary to maintain frequency
+            of every seq_id.
+
+        Atrributes:
+            id_type (int): Determines to which wiki convention it corresponds.
+
+        """
+        seq_id.strip(' ')
+        id_type = self.validate(seq_id)
+        if id_type == 0:
+            return 0
+        else:
+            if id_type == 1:
+                split_id = list(map(str, seq_id.split('/')))
+                if split_id[0] in seq_id_dict:
+                    seq_id_dict[split_id[0]] += 1
+                else:
+                    seq_id_dict[split_id[0]] = 1
+                return int(split_id[1])
+            elif id_type == 2:
+                split_id = list(map(str, seq_id.split()))
+                if split_id[0] in seq_id_dict:
+                    seq_id_dict[split_id[0]] += 1
+                else:
+                    seq_id_dict[split_id[0]] = 1
+                return int(split_id[1][0])
+
+    def result_from_count(self, count: list):
```

Comment:
Better to make use of the enumerator to make this easier to follow. Compare my solution.